### PR TITLE
Dockerfile: Set path and target via ENV

### PR DESCRIPTION
**Pull request by passy**
Set $PATH and $CARGO_TARGET_DIR via the ENV instruction. This allows us
to drop the path from all subsequent RUN and CMD invocations which is
quite nice.

By setting a custom $CARGO_TARGET_DIR we avoid trouble with local IDEs
and such that would otherwise share the same directory as the container,
blocking and invalidating each other.

Jira:^H^H^H^H^H Hahaha, this isn't Gerrit.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,14 +34,15 @@ RUN     apt-get install -qq ctags
...
-RUN    /root/.cargo/bin/cargo install rustfmt
-RUN    /root/.cargo/bin/cargo install cargo-watch
-RUN    /root/.cargo/bin/cargo install cargo-outdated
-RUN    /root/.cargo/bin/cargo install cargo-graph
-RUN    /root/.cargo/bin/cargo install cargo-modules
-RUN    /root/.cargo/bin/cargo install cargo-count
-
-RUN    /root/.cargo/bin/rustup install nightly
-RUN    /root/.cargo/bin/rustup run nightly cargo install clippy
+ENV    CARGO_TARGET_DIR targetdocker
```
Comment (daithiocrualaoich) [inline Dockerfile:37]: Do we need this in `.gitignore` also?

File: Dockerfile [hunk 2]
```diff
@@ -34,14 +34,15 @@ RUN     apt-get install -qq ctags
...
+ENV    PATH /root/.cargo/bin:$PATH
+
+RUN    cargo install rustfmt
```
Comment (daithiocrualaoich) [inline Dockerfile:40]: Wow... really nice what you made happen here.

File: Dockerfile (hunk 1 above)
Comment (passy) [inline Dockerfile:37]: Yes!

**Comment by daithiocrualaoich** [comment]
No. This is not Gerrit.

**Pull request merged**